src/api/rolesApi.tsx

```diff
@@ -105,14 +119,36 @@ export const useAddRole = () => {
 
 export const useDeleteRole = () => {
   const queryClient = useQueryClient();
+  const roles = useRoles();
   const { dispatchToast } = useToastController("toaster");
   return useMutation(
     ["deleteRole"],
     async (id: number) => {
-      return spWebContext.web.lists
+      let error = undefined;
+      const thisRole = roles.data?.find((role) => (role.Id = id));
```

Comment:
This is accidently setting it, instead of comparing it, so just returns the first one, which has your user id, so is looking at and removing you from the group

```suggestion
      const thisRole = roles.data?.find((role) => (role.Id === id));
```